lib/wasm/src/func_translator.rs

```diff
@@ -226,7 +231,7 @@ fn cur_srcloc(reader: &BinaryReader) -> ir::SourceLoc {
     // This will wrap around of a single function's byte code is larger than 4 GB, but a) the
     // WebAssembly format doesn't allow for that, and b) that would hit other Cranelift
     // implementation limits anyway.
-    ir::SourceLoc::new(reader.current_position() as u32)
+    ir::SourceLoc::new(reader.original_position() as u32)
```

Comment:
Should this be `reader.original_position() + reader.current_position()` to get the full offset to the current opcode?

Also, we should coordinate with SpiderMonkey to see whether changing from function-relative offsets to module offsets will affect things there. @bnjbvr Do you know if baldrdash has the function start offsets around when it calls `emit_metadata`? If so, it could use them to translate back to function-relative offsets if needed.

And we should update the comment here in `cur_srcloc` :-).

Reply:
`original_position()` already [adds](https://github.com/yurydelendik/wasmparser.rs/blob/master/src/binary_reader.rs#L123) offset (that is provided during `new_with_offset`)

Reply:
In `emit_metadata`, Baldrdash already saves only the function relative offset; the function offset base is added in the C++ code. So if we change this to use module-relative offsets instead, my understanding is we can just update the C++ code.

Reply:
Sounds good, let's merge it.